### PR TITLE
Add TopNavBar compose component and update toolbar title padding

### PR DESCRIPTION
**Pull request by marianeum**
+ [x] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-android/blob/main/CONTRIBUTING.md)

Remember to include the following changes:
+ [x] Component `README.md`
+ [x] Tests
+ [x] Docs (either update [backpack-docs](https://github.com/Skyscanner/backpack-docs) now, or create a follow up ticket)

If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/main/CODE_REVIEW_GUIDELINES.md)

### CODE REVIEW

File: Backpack/src/main/res/values/themes.xml [hunk 1]
```diff
@@ -9,6 +9,7 @@
     <item name="windowNoTitle">true</item>
     <item name="windowActionBar">false</item>
     <item name="colorButtonNormal">@android:color/white</item>
+    <item name="toolbarStyle">@style/BpkToolbarStyle</item>
```
Comment (marianeum) [inline Backpack/src/main/res/values/themes.xml:12]: This triggers an update of all docs screenshots

File: scripts/record_screenshot_tests.sh [hunk 2]
```diff
@@ -17,6 +17,6 @@
 
 set -e
 ./gradlew :app:recordOssDebugAndroidTestScreenshotTest -Pandroid.testInstrumentationRunnerArguments.variant=default
-./gradlew :app:recordOssDebugAndroidTestScreenshotTest -Pandroid.testInstrumentationRunnerArguments.variant=themed
-./gradlew :app:recordOssDebugAndroidTestScreenshotTest -Pandroid.testInstrumentationRunnerArguments.variant=rtl
 ./gradlew :app:recordOssDebugAndroidTestScreenshotTest -Pandroid.testInstrumentationRunnerArguments.variant=dm
+./gradlew :app:recordOssDebugAndroidTestScreenshotTest -Pandroid.testInstrumentationRunnerArguments.variant=rtl
```
Comment (marianeum) [inline scripts/record_screenshot_tests.sh:21]: changed the order for these as themed is the one we need least, so if you run all 4 you can cancel earlier

File: Backpack/src/main/res/values/themes.xml (hunk 1 above)
Comment (bvitaliyg) [reply Backpack/src/main/res/values/themes.xml:12]: This changed the spacing between the title and the nav icon. Have you tested all scrolling animations (including RTL) manually?

Comment (marianeum) [reply Backpack/src/main/res/values/themes.xml:12]: yep works fine in both

File: docs/compose/NavBar/README.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+```Kotlin
+import net.skyscanner.backpack.compose.icon.BpkIcon
+import net.skyscanner.backpack.compose.navigationbar.BpkTopNavBar
+import net.skyscanner.backpack.compose.navigationbar.NavIcon
+import net.skyscanner.backpack.compose.navigationbar.IconAction
+
+BpkTopNavBar(
+  title = stringResource(R.string.navigation_bar_title),
+  navIcon = NavIcon.Back(contentDescription = stringResource(R.string.navigation_back)) { /** onClick **/ },
+  actions = listOf(
```
Comment (bvitaliyg) [inline docs/compose/NavBar/README.md:58]: Lists :(

**Review by bvitaliyg — COMMENTED** [review]
Just few comments

File: backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/utils/Modifier.kt [hunk 4]
```diff
@@ -53,3 +59,17 @@ internal fun Modifier.autoMirror(enabled: Boolean = true): Modifier =
...
     }
   }
+
+internal fun Modifier.unboundClickable(role: Role? = null, onClick: () -> Unit): Modifier =
+  composed(
+    factory = {
+      clickable(
+        interactionSource = remember { MutableInteractionSource() },
+        indication = rememberRipple(bounded = false),
+        role = Role.Button,
```
[the remark this answers is not in the document]
Comment (marianeum) [reply backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/utils/Modifier.kt:69]: whoops

File: backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/utils/Modifier.kt [hunk 5]
```diff
@@ -53,3 +59,15 @@ internal fun Modifier.autoMirror(enabled: Boolean = true): Modifier =
       this@drawWithContent.drawContent()
     }
   }
+
+internal fun Modifier.unboundClickable(role: Role? = null, onClick: () -> Unit): Modifier =
```
Comment (bvitaliyg) [inline backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/utils/Modifier.kt:63]: What about doing clickable which mimics the Compose clickable but adds bounded param?

**Review by bvitaliyg — APPROVED** [review]
Approved :) Consider the clickable change as well :)

**Review by bvitaliyg — APPROVED** [review]

**Pull request merged**